### ISSUE TITLE
Unique index on a foreign key causes subsequent migrations to add and drop the index

### ISSUE THREAD

**Issue by joelverhagen** [aspnet/EntityFramework#6470]
### Steps to reproduce

1. Open the following solution in VS: [MigrationProblem.zip](https://github.com/aspnet/EntityFramework/files/454019/MigrationProblem.zip)
1. Notice there are no migrations.
1. Run `Add-Migration InitialMigration`.
1. Run `Add-Migration NoChangeMigration`.

### The issue

The second migration (`NoChangeMigration`) should have empty upgrade and downgrade methods, since no changes to the model happened. Instead, a weird migration is added that added and removes an index.

The problem seems to be related to the fact there is a unique index on a field that is also a foreign key.

The `NoChangeMigration` has this code:

```csharp
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore.Migrations;

namespace MigrationProblem.Migrations
{
    public partial class NoChangeMigration : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropIndex(
                name: "IX_CheckNotifications_CheckId",
                table: "CheckNotifications");

            migrationBuilder.CreateIndex(
                name: "IX_CheckNotifications_CheckId",
                table: "CheckNotifications",
                column: "CheckId",
                unique: true);
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropIndex(
                name: "IX_CheckNotifications_CheckId",
                table: "CheckNotifications");

            migrationBuilder.CreateIndex(
                name: "IX_CheckNotifications_CheckId",
                table: "CheckNotifications",
                column: "CheckId");
        }
    }
}
```

### Further technical details

EF Core version: 1.0.0
Operating system: Windows 10
Visual Studio version: VS 2015 Update 3

**Comment by bricelam** [comment]
This no longer repros on the dev branch (which will become version 1.1.0).

I think @AndriySvyryd fixed some things relating to indexes and foreign keys...

**Comment by bricelam** [comment]
Most likely fixed by #6294
